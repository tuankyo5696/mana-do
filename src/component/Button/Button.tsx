import React, { MouseEventHandler, CSSProperties, ButtonHTMLAttributes } from 'react';

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  children: string;
  onClick?: MouseEventHandler<HTMLButtonElement>;
  className?: string;
  disabled?: boolean;
  name?: string;
  type?: 'button' | 'submit' | 'reset';
  style?: CSSProperties;
}

const Button: React.FC<ButtonProps> = ({ children, onClick, className, style, disabled, ...args }) => {
  return (
    <button
      onClick={onClick}
      className={className}
      disabled={disabled}
      style={{ backgroundColor: '#337ab7', color: '#fff', ...style }}
      {...args}
    >
      {children}
    </button>
  );
};

Button.defaultProps = {
  style: {},
  className: '',
  type: 'button',
  disabled: false
};

export default Button;
