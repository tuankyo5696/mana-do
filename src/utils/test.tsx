import React from 'react';
import renderer from 'react-test-renderer';

interface Config {
  props: object;
  description: string;
}

const singleSnapTest = (tree: any, description: string) => {
  test(description, () => {
    expect(tree).toMatchSnapshot();
  });
};

const testSnapshots = (Component: any, configs: Config[]) => {
  describe('snapshots', () => {
    configs.forEach((config) => {
      const { props, description } = config;
      const tree = renderer.create(<Component {...props} />).toJSON() 
      singleSnapTest(tree, description);
    });
  });
};

export { testSnapshots };
