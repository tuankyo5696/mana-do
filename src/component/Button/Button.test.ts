import { testSnapshots } from '../../utils/test';
import Button from './Button';
describe('render Button', () => {
  testSnapshots(Button, [
    {
      props: {
        children: 'Button'
      },
      description: 'render default Button'
    },
    {
      props: {
        children: 'Button disabled'
      },
      description: 'render disabled Button'
    }
  ]);
});

