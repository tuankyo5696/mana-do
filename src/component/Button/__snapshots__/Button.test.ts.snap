// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Button snapshots render default Button 1`] = `
<button
  className=""
  disabled={false}
  style={
    Object {
      "backgroundColor": "#337ab7",
      "color": "#fff",
    }
  }
  type="button"
>
  Button
</button>
`;

exports[`render Button snapshots render disabled Button 1`] = `
<button
  className=""
  disabled={false}
  style={
    Object {
      "backgroundColor": "#337ab7",
      "color": "#fff",
    }
  }
  type="button"
>
  Button disabled
</button>
`;
